jlm/llvm/opt/alias-analyses/Andersen.hpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023 Håvard Krogstie <krogstie.havard@gmail.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_LLVM_OPT_ALIAS_ANALYSES_ANDERSEN_HPP
+#define JLM_LLVM_OPT_ALIAS_ANALYSES_ANDERSEN_HPP
+
+#include <jlm/llvm/ir/operators.hpp>
+#include <jlm/llvm/ir/RvsdgModule.hpp>
+#include <jlm/llvm/opt/alias-analyses/AliasAnalysis.hpp>
+#include <jlm/llvm/opt/alias-analyses/PointerObjectSet.hpp>
+
+namespace jlm::llvm::aa
+{
+
+/**
+ * class implementing Andersen's set constraint based alias analysis.
+ * The analysis is inter-procedural, field-insensitive,
+ * context-insensitive, flow-insensitive, and uses a static heap model.
+ */
+class Andersen final : public AliasAnalysis
```

Comment:
Could you provide a reference to the paper such that people know which one it is that you implemented? Similarly to [Steensgaard](https://github.com/phate/jlm/blob/f2e1a788b57d40b0438223e195d9afd776b1dbd8/jlm/llvm/opt/alias-analyses/Steensgaard.hpp#L15C1-L23C47).